models/model.py

```diff
@@ -98,20 +99,26 @@ def get_mode_index(self):
         pass
 
     def update_pack_temp_data(self) -> None:
+        # if we already have 600 values, pop the last one and then insert the newest value at zero
         if len(self.pack_temp_data) >= 600:
             self.pack_temp_data.pop()
         self.pack_temp_data.insert(0, self.get_pack_temp())
 
     def add_pinned_data(self, id: int) -> None:
-        self.pinned_data[id] = DebugPlotValue(name=DATA_IDS[id]['name'],
-                                              data=[round(self.current_data[id], 1) if self.current_data[id] is not None else self.current_data[id]], unit=DATA_IDS[id]['units'])
+        # add the given id to all pinned data
+        data = round(self.current_data[id], 1) if self.current_data[id] is not None else self.current_data[id]
```

Comment:
since you access self.current_data[id] three times in this one line it might be better to just declare it the line before for readability